src/web/services/course.service.ts

```diff
@@ -25,6 +25,17 @@ export class CourseService {
     return this.httpRequestService.get('/course', paramMap);
   }
 
+  /**
+   * Gets all course data for an instructor by calling API.
+   */
+  getCoursesForInstructor(entityType: string, courseStatus: string): Observable<Courses> {
```

Comment:
If it is for instructor, the entityType would always be `instructor` right?

Reply:
Yup realised. I have changed the function and made it more general. Both students and instructors can now use this to get all courses.